tests/contrib/requests/test_requests_distributed.py

```diff
@@ -0,0 +1,116 @@
+
+# 3p
+from nose.tools import eq_, assert_not_equal
+from threading import Lock, Thread
+from sys import version_info
+if version_info[0] < 3:
+    from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
+else:
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+from time import sleep
+
+# project
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.ext import http, errors
+from tests.test_tracer import get_dummy_tracer
+from .test_requests import get_traced_session
+
+# host/port on which our dumb server is listening
+_SERVER_ADDRESS = ('localhost', 8082)
+
+_lock = Lock()
+_running = None
+_client_tracer, _session = get_traced_session()
+_server_tracer = get_dummy_tracer()
+_httpd = None
+
+class DummyServer(BaseHTTPRequestHandler):
```

Comment:
Why do we need this complexity? We can simply Mock the request call and ensures that sent headers are the one we are expecting (so the set values / types).

Reply:
OK, will rework the test to downgrade it to a simple mock, NP.